src/Functions/makeDate.cpp

```diff
@@ -0,0 +1,170 @@
+            DataTypePtr argument_type = arguments[i];
+            if (!isNumber(argument_type) && !isStringOrFixedString(argument_type))
+                throw Exception("Argument '" + std::string(argument_names[i]) + "' for function " + getName() + " must be number",
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+
+        return std::make_shared<typename Traits::ReturnDataType>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        const DataTypePtr converted_argument_type = std::make_shared<DataTypeFloat64>();
+        Columns converted_arguments;
+        for (size_t i = 0; i < arguments.size(); ++i)
+        {
+            const auto & argument_type = arguments[i].type;
+            ColumnPtr argument_column = arguments[i].column->convertToFullColumnIfConst();
+
+            if (converted_argument_type->equals(*argument_type))
+                converted_arguments.push_back(argument_column);
+            else if (isNumber(argument_type))
+                converted_arguments.push_back(convert_to_number_func.executeImpl(
+                    {ColumnWithTypeAndName(argument_column, argument_type, "")}, converted_argument_type, input_rows_count));
+            else if (isStringOrFixedString(argument_type))
+                converted_arguments.push_back(parse_number_func.executeImpl(
+                    {ColumnWithTypeAndName(argument_column, argument_type, "")}, converted_argument_type, input_rows_count));
+            else
+                throw Exception("Argument '" + std::string(argument_names[i]) + "' for function " + getName() + " must be number",
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+
+        auto res_column = Traits::ReturnColumnType::create(input_rows_count);
+        auto & result_data = res_column->getData();
+
+        const auto & year_data = typeid_cast<const ColumnFloat64 &>(*converted_arguments[0]).getData();
+        const auto & month_data = typeid_cast<const ColumnFloat64 &>(*converted_arguments[1]).getData();
+        const auto & day_data = typeid_cast<const ColumnFloat64 &>(*converted_arguments[2]).getData();
+
+        const auto & date_lut = DateLUT::instance();
+
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            const auto year = year_data[i];
+            const auto month = month_data[i];
+            const auto day = day_data[i];
+
+            Int32 day_num = 0;
+
+            if (year >= Traits::MIN_YEAR &&
+                year <= Traits::MAX_YEAR &&
+                month <= 12 &&
+                day <= 31 &&
+                YearMonthDayToSingleInt(year, month, day) <= Traits::MAX_DATE)
+            {
+                day_num = date_lut.makeDayNum(year, month, day);
+            }
+
+            result_data[i] = day_num;
+        }
+
+        return res_column;
+    }
+
+private:
+    FunctionToFloat64 convert_to_number_func;
```

Comment:
Do we really need Float64? What will happen if we replace it with Float32 or even Int16 (at least overflow issues)?